component/Page/Profile/index.tsx

```diff
@@ -58,27 +61,31 @@ export default function ProfilePage(props: PageProps) {
 
     const { copyToClipboardToast } = useToastHook()
 
-    const showDownloadToast = () => toast('Download the Helio app now to create a profile and send incognito messages', {
-        position: "bottom-center",
-        autoClose: 3000,
-        hideProgressBar: true,
-        closeOnClick: true,
-        pauseOnHover: true,
-        theme: "dark",
-        closeButton: false,
-        style: {
-            borderRadius: '8px',
-            width: '270px',
-            margin: 'auto',
-            marginBottom: '60px',
-            textAlign: 'center',
-            backgroundColor: 'rgba(0, 0, 0, 0.25)'
-        },
-    })
+    const showDownloadToast = (eventName?: BetterSocialEventTracking) => {
+        eventName && sendAnalytics(eventName)
+        toast('Download the Helio app now to create a profile and send incognito messages', {
+            position: "bottom-center",
+            autoClose: 3000,
+            hideProgressBar: true,
+            closeOnClick: true,
+            pauseOnHover: true,
+            theme: "dark",
+            closeButton: false,
+            style: {
+                borderRadius: '8px',
+                width: '270px',
+                margin: 'auto',
+                marginBottom: '60px',
+                textAlign: 'center',
+                backgroundColor: 'rgba(0, 0, 0, 0.25)'
+            },
+        })
+    }
 
     const onShareButtonClicked = () => {
         navigator.clipboard.writeText(`${publicRuntimeConfig.DYNAMIC_LINK_DOMAIN}/${username}`)
         copyToClipboardToast()
+        sendAnalytics(BetterSocialEventTracking.PROFILE_SCREEN_SHARE_BUTTON_CLICKED)
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `onShareButtonClicked` function is missing error handling for the `navigator.clipboard.writeText` promise. If this fails (for example, due to permission issues), it will throw an unhandled promise rejection. Consider adding a try-catch block to handle potential errors.
```diff
-     const onShareButtonClicked = () => {
-         navigator.clipboard.writeText(`${publicRuntimeConfig.DYNAMIC_LINK_DOMAIN}/${username}`)
-         copyToClipboardToast()
-        sendAnalytics(BetterSocialEventTracking.PROFILE_SCREEN_SHARE_BUTTON_CLICKED)
-    }
+     const onShareButtonClicked = async () => {
+         try {
+             await navigator.clipboard.writeText(`${publicRuntimeConfig.DYNAMIC_LINK_DOMAIN}/${username}`)
+             copyToClipboardToast()
+             sendAnalytics(BetterSocialEventTracking.PROFILE_SCREEN_SHARE_BUTTON_CLICKED)
+         } catch (error) {
+             console.error('Failed to write to clipboard:', error)
+         }
+     }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->